### PR TITLE
fix(cloudfront-signer): accept passphrase when signing

### PR DESCRIPTION
**Pull request by unchaptered**
getSIgnedUrl need new params of passphrase of RSA Key.

### Issue

- Issue #4231

### Description

Add new params of passphrase to fix issue.

### Testing

Add new sample of passpharse to fix issue.
But, in testcode, aws have string type of privateKey.
So, I sugggest new privateKey and passphrase from AWS officials.

### Additional context

Please consider its.
Now, getSignedUrl isn't working...

### CODE REVIEW

File: packages/cloudfront-signer/src/sign.ts [hunk 1]
```diff
@@ -327,14 +343,20 @@ class CloudfrontSignBuilder {
     };
   }
 
-  private signData(data: string, privateKey: string | Buffer): string {
+  private signData(data: string, privateKey: string | Buffer, passphrase: string): string {
```
Comment (trivikr) [inline packages/cloudfront-signer/src/sign.ts:346]:
```suggestion
  private signData(data: string, privateKey: string | Buffer, passphrase?: string): string {
```

File: packages/cloudfront-signer/src/sign.ts [hunk 2]
```diff
@@ -327,14 +343,20 @@ class CloudfrontSignBuilder {
...
+        key: privateKey,
+        passphrase: passphrase,
+      },
+      "base64"
+    );
   }
 
-  private signPolicy(policy: string, privateKey: string | Buffer): string {
-    return this.normalizeBase64(this.signData(policy, privateKey));
+  private signPolicy(policy: string, privateKey: string | Buffer, passphrase: string): string {
```
Comment (trivikr) [inline packages/cloudfront-signer/src/sign.ts:358]:
```suggestion
  private signPolicy(policy: string, privateKey: string | Buffer, passphrase?: string): string {
```

File: packages/cloudfront-signer/src/sign.ts [hunk 3]
```diff
@@ -327,14 +343,20 @@ class CloudfrontSignBuilder {
...
     const sign = createSign("RSA-SHA1");
     sign.update(data);
-    return sign.sign(privateKey, "base64");
+    return sign.sign(
+      {
+        key: privateKey,
+        passphrase: passphrase,
+      },
+      "base64"
+    );
```
Comment (trivikr) [inline packages/cloudfront-signer/src/sign.ts:355]: nit: can be shortened
```suggestion
    return sign.sign({ key: privateKey, passphrase }, "base64");
```

The passphrase is optional in PrivateKeyInput, so we don't have to do undefined check

**Review by trivikr — CHANGES_REQUESTED** [review]

File: packages/cloudfront-signer/src/sign.ts [hunk 4]
```diff
@@ -11,6 +11,8 @@ export interface CloudfrontSignInputBase {
   keyPairId: string;
   /** The content of the Cloudfront private key. */
   privateKey: string | Buffer;
+  /** The passphrase of RSA-SHA1 key*/
+  passphrase: string;
```
[the remark this answers is not in the document]
Comment (unchaptered) [reply packages/cloudfront-signer/src/sign.ts]: Thanks to your kind guide.
I've never consider that about maintaing backward compatibility.


Thanks! :)

File: packages/cloudfront-signer/src/sign.ts (hunk 3 above)
Comment (unchaptered) [reply packages/cloudfront-signer/src/sign.ts]: Oh, thanks for your recommendation.

**Review by unchaptered — COMMENTED** [review]
Sorry to late response.
Maybe, I  don't know if I understand open source culture properly.

After chang codes as your requests, am I commit it with mesion to you?

@trivikr

**Review by unchaptered — COMMENTED** [review]
Sorry to late response.
Maybe, I  don't know if I understand open source culture properly.

After chang codes as your requests, am I commit it with mesion to you?

@trivikr

**Review by trivikr — APPROVED** [review]

**Pull request merged**